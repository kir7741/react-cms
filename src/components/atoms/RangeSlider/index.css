.range-slider {
	display: grid;
	align-items: center;
	position: relative;
	margin-bottom: 10px;
	padding-bottom: 1.25em;
	min-height: 60px;

	& input {
		appearance: none;
		width: 100%;
		height: 15px;
		background: #fff;
		outline: none;
		border-radius: 8px;
		transition: opacity .2s;
	}

	& input:hover {
		opacity: 1;
	}

	& input::-webkit-slider-thumb {
		appearance: none;
		width: 25px;
		height: 25px;
		background: var(--dark-green);
		border-radius: 50%;
		cursor: pointer;
	}

	& input::-moz-range-thumb {
		width: 25px;
		height: 25px;
		background: var(--dark-green);
		border-radius: 50%;
		cursor: pointer;
	}

	& span {
		position: absolute;
		bottom: 0px;
		font-size: 14px;
		color: var(--error-red);
	}

}
