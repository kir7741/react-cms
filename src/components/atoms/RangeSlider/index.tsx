import React, { InputHTMLAttributes } from 'react';
import classnames from 'classnames';

import styles from './index.css';

interface RangeSliderProperty extends InputHTMLAttributes<HTMLInputElement> {

	/**
	 * 樣式設定
	 *
	 * @type {string}
	 * @memberof RangeSliderProperty
	 */
	className?: string;

	/**
	 * 更新表單檢核狀態
	 *
	 * @memberof RangeSliderProperty
	 */
	updateCtrlValidity?: () => void;

	/**
	 * change 時觸發的函式
	 *
	 * @memberof RangeSliderProperty
	 */
	onChangeValue: (val: string) => void;

}

const RangeSlider: React.FC<RangeSliderProperty> = ({
	className,
	min = 0,
	max = 100,
	value = 0,
	step = 0,
	updateCtrlValidity = () => {},
	onChangeValue = () => {}

}) => {

	const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
		e.preventDefault();
		const val = e.target.value;
		onChangeValue(val);
	};

	return (
		<div className={classnames(styles.rangeSlider, className)}>
			<input
				type="range"
				min={min}
				max={max}
				value={value}
				step={step}
				onChange={e => handleInputChange(e)}
			/>
		</div>
	);
};

export default RangeSlider;
