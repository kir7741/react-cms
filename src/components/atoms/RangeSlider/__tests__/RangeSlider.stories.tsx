import React from 'react';
import { storiesOf } from '@storybook/react';

import RangeSlider from 'components/atoms/RangeSlider';

const stories = storiesOf('atoms/RangeSlider', module);

stories.add('__interactive', () => <RangeSlider />);
