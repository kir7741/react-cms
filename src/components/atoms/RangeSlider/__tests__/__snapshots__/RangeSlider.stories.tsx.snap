// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="rangeSlider"
  >
    <input
      max="100"
      min="0"
      step="0"
      type="range"
      value="0"
    />
  </div>
</DocumentFragment>
`;
